### PR TITLE
script/run_mypy: Support mypy 0.740

### PR DESCRIPTION
**Pull request by sebastian-philipp**
mypy 0.740 changed the output, thus we have to adopt to it.

Relates to: https://github.com/python/mypy/issues/7806

Signed-off-by: Sebastian Wagner <sebastian.wagner@suse.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`

</details>

### CODE REVIEW

File: src/script/run_mypy.sh [hunk 1]
```diff
@@ -3,11 +3,13 @@
...
 # generates a report at src/mypy_report.txt
 
+set -e
+
 python3 -m venv .mypy_venv
 
 . .mypy_venv/bin/activate
 
-pip install $(find * -name requirements.txt | grep -v node_modules | awk '{print "-r  " $0}')
+! pip install $(find * -name requirements.txt | grep -v node_modules | awk '{print "-r  " $0}')
```
Comment (epuertat) [inline src/script/run_mypy.sh]:
```suggestion
! pip install $(find -name requirements.txt -not -path './frontend/*' -printf '-r%p ')
```

**Comment by sebastian-philipp** [comment]
> Would it make sense to move mypy to (in the case of dashboard) `tox` environment? I saw there are pytest plugins to run mypy: https://pypi.org/project/pytest-mypy/ and https://pypi.org/project/pytest-mypy-plugins/

Yes! As I did in #31071 !

**Comment by epuertat** [comment]
> > Would it make sense to move mypy to (in the case of dashboard) `tox` environment? I saw there are pytest plugins to run mypy: https://pypi.org/project/pytest-mypy/ and https://pypi.org/project/pytest-mypy-plugins/
> 
> Yes! As I did in #31071 !

Thanks, I missed that one! But that only covers python-common pkg, right? So we should do the same for dashboard one.

**Comment by sebastian-philipp** [comment]
> > Yes! As I did in #31071 !
> 
> Thanks, I missed that one! But that only covers python-common pkg, right? So we should do the same for dashboard one.

First step into this direction would be to have a detailed look into

https://gist.github.com/sebastian-philipp/25f70aae3b0d21b1a781c110a7ef8be4#file-mypy_report-txt-L400-L1043

**Comment by epuertat** [comment]
> > > Yes! As I did in #31071 !
> > 
> > 
> > Thanks, I missed that one! But that only covers python-common pkg, right? So we should do the same for dashboard one.
> 
> First step into this direction would be to have a detailed look into
> 
> https://gist.github.com/sebastian-philipp/25f70aae3b0d21b1a781c110a7ef8be4#file-mypy_report-txt-L400-L1043
> 
> I see this script as a path towards proper mypy integration into all components. As soon as we have this, we can simply remove this one.

And wouldn't be mypy smart enough to apply the hints it provided? :sweat_smile: 

I'm wondering how we could enforce the incremental python typing without breaking/becoming a pain... Maybe creating an (optional) Jenkins job that runs the mypy toxenv and checks against already reported errors?

**Comment by sebastian-philipp** [comment]
> And wouldn't be mypy smart enough to apply the hints it provided? 

not really.

> 
> I'm wondering how we could enforce the incremental python typing without breaking/becoming a pain...

I don't think so. 

> Maybe creating an (optional) Jenkins job that runs the mypy toxenv and checks against already reported errors?

after doing the mypy runs for a few months, my impression is: either we have it integrated into tox, or we run it completely manual.

**Pull request merged**